### ISSUE TITLE
Populate diffusion-repo-prefix-list automatically

### ISSUE THREAD

**Issue by colonelpanic8** [ajtulloch/phabricator.el#3]
@ajtulloch do you know if there is some way to handle this automatically?

**Comment by ajtulloch** [comment]
Yeah, I think we can actually get it from the `.arcconfig` file in the project root - does that work for you folks at Uber? I'm not sure what's FB-specific and what is available in OSS Phabricator.

Like, if the .arcconfig contains:

```
{ 
  "conduit_uri": "https://phabricator.example.com/api/",
  "project_id":  "CALLSIGN",
  ...
}
```

then we can just easily map a path of `a/b/c` to 

```
# http query path join, but w/e
os.path.join(
  dirname(<conduit_uri>), 
  "diffusion", 
  <callsign>,
  "browse",
  "master",
  `a/b/c`
)
```

or something? I'm not sure what the relationship between the `project_id`/`project.name` field and the callsign for a repository is, and it doesn't look like there's a way to find the callsign for a given project in Conduit? https://secure.phabricator.com/conduit/

**Comment by colonelpanic8** [comment]
> Yeah, I think we can actually get it from the .arcconfig file in the project root - does that work for you folks at Uber? I'm not sure what's FB-specific and what is available in OSS Phabricator.

This is what I tried to do when I implemented this myself -- unfortunately, we have quite a few repositories for which project_id is not correctly populated. Still, Looking at .arcconfig is probably better than nothing.

> Like, if the .arcconfig contains:
> 
> { 
>   "conduit_uri": "https://phabricator.example.com/api/",
>   "project_id":  "CALLSIGN",
>   ...
> }
> then we can just easily map a path of a/b/c to
> 
> # http query path join, but w/e
> 
> os.path.join(
>   dirname(<conduit_uri>), 
>   "diffusion", 
>   <callsign>,
>   "browse",
>   "master",
>   `a/b/c`
> )
> or something? I'm not sure what the relationship between the project_id/project.name field and the callsign for a repository is, and
> it doesn't look like there's a way to find the callsign for a given project in Conduit? https://secure.phabricator.com/conduit/

Yep, I looked in to that as well. A conduit url for resolving this mapping seems like the best solution. Could you possibly prod the right folks at facebook to get this to happen?
